refactor(core): deprecate `WrappedValue` (#36819)
angular/angular: aio/content/guide/deprecations.md

For more information about the npm package format, see the [Angular Package Format spec](https://goo.gl/jB3GVv).

{@a wrapped-value}

> ### Drop support for detecting changes in `async` pipe if `Observable` returns the same instance

Reviewer: The format we've been using with other sections is just naming the thing that is deprecated, sans something like "drop support"

```suggestion
###  `WrappedValue` 
```

> The purpose of the `WrappedValue` is to allow same object instance to be treated as different for the purposes of change detection.

Reviewer: ```suggestion
The purpose of `WrappedValue` is to allow the same object instance to be treated as different for the purposes of change detection.
```

> It is currently used with `async` pipe and only with `Observables`.

Reviewer: ```suggestion
It is commonly used with `async` pipe in the case where the `Observable` produces the same instance of the value.
```

> The use case which it covers is if the `Observable` produces the same instance of the value but it is desirable to still try to mark it as changed for the purposes of change detection. 

Reviewer 2: Maybe mention the alternatives for those who were depending on the `WrappedValue`? A couple options I can think of:
* changing the primitive to an object and change the reference
* subscribing to the observable and calling `markForCheck` when you want CD to run

Author: done

> The use case which it covers is if the `Observable` produces the same instance of the value but it is desirable to still try to mark it as changed for the purposes of change detection. 

Reviewer: ```suggestion
```

> We believe tha the above use case is too rare to warrant special handling in the framework.
> (Having special handling causes application slowdown for the users and mental load for the developers.)

Reviewer: ```suggestion
Given that this use case is relatively rare and special handling impacts application performance, we have deprecated it in v10.
```

> No replacement is planned for this deprecation.

Reviewer: Can we add a few sentences about how to replace `WrappedValue` with your own code if you're currently depending on it?

Author: Done

For more information about the npm package format, see the [Angular Package Format spec](https://goo.gl/jB3GVv).

{@a wrapped-value}
###  `WrappedValue` 

The purpose of `WrappedValue` is to allow the same object instance to be treated as different for the purposes of change detection.

> It is commonly used with `async` pipe in the case where the `Observable` produces the same instance of the value.

Reviewer: ```suggestion
It is commonly used with the `async` pipe in the case where the `Observable` produces the same instance of the value.
```

Given that this use case is relatively rare and special handling impacts application performance, we have deprecated it in v10.
No replacement is planned for this deprecation.

> If you rely on the behavior that same object instance should cause change detection you have two options:

Reviewer: ```suggestion
If you rely on the behavior that the same object instance should cause change detection, you have two options:
```

> - Clone the resulting value so that it has new identity.

Reviewer: ```suggestion
- Clone the resulting value so that it has a new identity.
```
